plugins/vault-backend/src/service/plugin.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import {
+  VaultApi,
+  VaultExtensionPoint,
+  vaultExtensionPoint,
+} from '@backstage/plugin-vault-node';
+import { VaultBuilder } from './VaultBuilder';
+
+class VaultExtensionPointImpl implements VaultExtensionPoint {
+  vaultClient?: VaultApi;
+  tokenRenewEnabled: boolean = false;
+
+  setClient(vaultClient: VaultApi): void {
+    if (this.vaultClient) {
+      throw new Error('The vault client has been already set');
+    }
+    this.vaultClient = vaultClient;
+  }
+
+  enableTokenRenew(): void {
+    this.tokenRenewEnabled = true;
+  }
+}
+
+/**
+ * Vault backend plugin
+ *
+ * @public
+ */
+export const vaultPlugin = createBackendPlugin({
+  pluginId: 'vault',
+  register(env) {
+    const vaultExtensions = new VaultExtensionPointImpl();
+    env.registerExtensionPoint(vaultExtensionPoint, vaultExtensions);
+
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        config: coreServices.rootConfig,
+        scheduler: coreServices.scheduler,
+        httpRouter: coreServices.httpRouter,
+      },
+      async init({ logger, config, scheduler, httpRouter }) {
+        let builder = VaultBuilder.createBuilder({
+          logger,
+          config,
+          scheduler,
+        });
+
+        if (vaultExtensions.vaultClient) {
+          builder = builder.setVaultClient(vaultExtensions.vaultClient);
+        }
+        if (vaultExtensions.tokenRenewEnabled) {
```

Comment:
I think it'd make sense to make this completely config driven so that you don't need to enable it via an extensions point as that's a bit more involved. We could extend the `schedule` config to accept either a schedule definition or a boolean, where `true` means we'll use the default schedule.